reflect/src/main/java/dagger/reflect/ReflectiveModuleParser.java

```diff
@@ -72,7 +84,12 @@ static void parse(Class<?> moduleClass, @Nullable Object instance,
 
         graphBuilder.add(key, binding);
       }
-      target = target.getSuperclass();
+
+      Class<?> superclass = target.getSuperclass();
+      if (superclass != Object.class && superclass != null) {
+        queue.add(superclass);
+      }
+      Collections.addAll(queue, target.getInterfaces());
```

Comment:
would it worth testing multiple inheritance?
i.e. `abstract class Foo extends Bar implements Baz, Quz`

Reply:
Added a test for this by way of testing that when an interface is included twice via a base class that it still works

Reply:
I was going for more the multiple interface to make sure all of the interfaces are checked on a single class not just the first one, but up to you